### PR TITLE
[AppBar] Test the UINavigationController accessibilityPerformEscape behavior.

### PR DESCRIPTION
**Pull request by jverkoey**
Pre-work for https://github.com/material-components/material-components-ios/issues/9576

### CODE REVIEW

File: components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+import MaterialComponents.MaterialAppBar
+
+class AppBarNavigationControllerAccessibilityPerformEscapeTests: XCTestCase {
```
Comment (codeman7) [inline components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift:21]: Should we document this class?

File: components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+  override func tearDown() {
+    appBarViewController = nil
+
+    super.tearDown()
+  }
+
+  // MARK: UINavigationController [ MDCAppBarViewController ]
+
+  func testInitializedWithAppBarViewController() {
```
Comment (codeman7) [inline components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift:39]: Should we add the expected test outcome to the name here and throughout?

File: components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    // When
+    appBarViewController.accessibilityPerformEscape()
+
+    // Then
+    XCTAssertEqual(navigationController.children.count, 2)
+  }
+
+  // MARK: UINavigationController [ UIViewController [ MDCAppBarViewController ] ]
+
+  func testInitializedWithAppBarViewControllerChild() {
```
Comment (codeman7) [inline components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift]: Nit: `testInitializedWithAppBarViewControllerChildOfViewController` the phrasing `...Child` seems ambiguous.

File: components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift (hunk 1 above)
Comment (codeman7) [inline components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift:21]: Should the class be prefixed? If so please update the file name as well.

**Review by codeman7 — APPROVED** [review]

File: components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift (hunk 2 above)
Comment (jverkoey) [reply components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift:39]: That the navigation controller is initialized with the app bar view controller is the expected outcome here. Did you interpret this test differently?

File: components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift (hunk 1 above)
Comment (jverkoey) [reply components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift:21]: The class name is aligned with the existing convention in this directory, so keeping it as-is is within the scope of this PR.

Comment (jverkoey) [reply components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift:21]: Done.

File: components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift (hunk 3 above)
Comment (jverkoey) [reply components/AppBar/tests/unit/AppBarNavigationControllerAccessibilityPerformEscapeTests.swift]: Adjusted. Please take a look again.

**Comment by jverkoey** [comment]
Made a few naming + docs changes. PTAL again @codeman7.

**Review by fluxminds — APPROVED** [review]

**Pull request merged**